### PR TITLE
Flies and Time Flies, but with a swarming component and a newfound hope.

### PR DESCRIPTION
**Pull request by ghost**
:cl: armhulen
add: flies!
add: when exposed to radiation, time flies!
/:cl:

The janitor is a very important job because when he doesn't clean the station, the bodies blood guts and gore end up lowering everyone's sanity and they start tiding hard, right? so lets amplify that effect with swarms of flies gathering around dead bodies that stay long enough. if those flies end up staying all the way through a rad storm then yeah you're gonna get some time flies that canonically power the ted device

also time flies is a really awesome pun, man.

### CODE REVIEW

**Comment by Ordonis** [comment]
I am all for additional pests. If this works well, I think adding additional vermin could actually make the Janitor's job engaging.

**Comment by davethwave** [comment]
Try adding a moodlet for this. If a crew member saw a pest onboard they would be at least annoyed right?

**Comment by Armhulen** [comment]
moooooodlets >:( but sure yeah

**Comment by Anonmare** [comment]
Time flies when you're having fun

**Comment by ExcessiveUseOfCobblestone** [comment]
@Armhulen just use the beauty component

File: code/modules/mob/living/simple_animal/friendly/cockroach.dm [hunk 1]
```diff
@@ -60,3 +60,152 @@
...
+	density = FALSE
+	mob_size = MOB_SIZE_TINY
+	mob_biotypes = list(MOB_ORGANIC, MOB_BUG)
+	movement_type = FLYING
+	gold_core_spawnable = HOSTILE_SPAWN
+	search_objects = 1 //have to find those plant trays!
+
+/mob/living/simple_animal/hostile/fly/Initialize()
+	. = ..()
+	AddComponent(/datum/component/swarming)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/friendly/cockroach.dm]: AddComponent(/datum/component/beauty, -`amount of ur choosing`)

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/friendly/cockroach.dm]: thaaaaaaaaank uuuuuuuu

File: code/modules/mob/living/simple_animal/friendly/cockroach.dm [hunk 2]
```diff
@@ -60,3 +60,152 @@
...
+		return
+	var/mob/living/simple_animal/hostile/fly/time/TF = owner
+	if(TF.twarp)
+		TF.twarp.cast(TF, TF)
+
+/obj/effect/proc_holder/spell/targeted/timewarp
+	name = "Time Warp!"
+	desc = "Warps you 10 seconds into the future! Whether you want to or not editon."
+	charge_max = 0
+	clothes_req = 0
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/cockroach.dm]: false

File: code/modules/mob/living/simple_animal/friendly/cockroach.dm [hunk 3]
```diff
@@ -60,3 +60,152 @@
...
+	phase_allowed = 1
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/cockroach.dm]: true

File: code/modules/mob/living/simple_animal/friendly/cockroach.dm [hunk 4]
```diff
@@ -60,3 +60,152 @@
...
+	selection_type = "range"
+	range = -1
+	include_user = 1
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/cockroach.dm]: true

File: code/modules/mob/living/simple_animal/friendly/cockroach.dm [hunk 5]
```diff
@@ -60,3 +60,152 @@
...
+	action_icon = 'icons/mob/actions/actions_minor_antag.dmi'
+	action_icon_state = "ninja_cloak"
+	action_background_icon_state = "bg_alien"
+
+/obj/effect/proc_holder/spell/targeted/timewarp/cast(list/targets,mob/living/user = usr)
+	playsound(get_turf(user), 'sound/magic/ethereal_enter.ogg', 50, 1, -1)
+	visible_message("<span class='boldwarning'>[targets] is warped 10 seconds into the future!</span>")
+	user.SetStun(0, FALSE)
+	user.SetKnockdown(0, FALSE)
+	user.setStaminaLoss(0, 0)
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/cockroach.dm]: setStaminaLoss(0, FALSE)

File: code/modules/mob/living/simple_animal/friendly/cockroach.dm [hunk 6]
```diff
@@ -60,3 +60,152 @@
...
+	var/obj/effect/dummy/timewarp/T2 = new(get_turf(user.loc))
+	user.forceMove(T2)
+	T2.jaunter = user
+	addtimer(CALLBACK(T2, /obj/effect/dummy/timewarp.proc/end_jaunt), 100)
+
+/obj/effect/dummy/timewarp
+	name = "distortion in spacetime"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "nothing"
+	var/canmove = 0
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/cockroach.dm]: false

File: code/modules/mob/living/simple_animal/friendly/cockroach.dm [hunk 7]
```diff
@@ -60,3 +60,152 @@
...
+	density = FALSE
+	anchored = TRUE
+	invisibility = 60
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+
+/obj/effect/dummy/timewarp/proc/end_jaunt()
+	if(jaunter)
+		visible_message("<span class='boldwarning'>[jaunter] arrives from the past!</span>")
+		jaunter.forceMove(get_turf(src))
+		playsound(get_turf(jaunter), 'sound/magic/ethereal_exit.ogg', 50, 1, -1)
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/cockroach.dm]: should be able to playsound from src here

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
I think there are other items that have been copy-pasted so if you could allow those to inherit that would be good as well.

**Comment by Armhulen** [comment]
no actually not my fault oranges refuses to tell me how to unfuck my git and this branch is completely broken now so ugh

**Comment by ninjanomnom** [comment]
git gud

**Pull request closed without merging**